### PR TITLE
chat-hook: send invites to all unsubscribed members of unmanaged chats

### PR DESCRIPTION
**Pull request by tacryt-socryp**
Send invites to all unsubscribed members of unmanaged chats to fix deleted DMs from metadata-store bug.

### CODE REVIEW

**Comment by tacryt-socryp** [comment]
NOTE: Have tested this and it works locally.

File: pkg/arvo/app/chat-hook.hoon [hunk 1]
```diff
@@ -83,8 +85,33 @@
...
+    ?:  ?=(%7 -.old)
+      =/  subscribers=(jug path ship)
+        %+  roll  ~(val by sup.bol)
+        |=  [[=ship =path] out=(jug path ship)]
+        ::  /(mailbox|backlog)/~ship/resource.name
+        ::
+        ?.  ?=([@ @ @ *] path)  out
+        =/  pax=^path  [i.t.path i.t.t.path ~]
+        (~(put ju out) pax ship)
+      =/  group  ~(. grpl bol)
```
Comment (Fang-) [inline pkg/arvo/app/chat-hook.hoon:99]: Could've just `grp:cc`. (;

**Review by Fang- — APPROVED** [review]
Ok, now it looks good to me.

**Comment by Fang-** [comment]
Just received this OTA. Apparently ~palfun hosts chat(s?) with no group associated with it.

https://github.com/urbit/urbit/blob/e17e508379fac7b87a331f15d47bdd72aeec6dbf/pkg/arvo/app/chat-hook.hoon#L912-L918

```
%load-failed
/app/chat-hook/hoon:<[917 3].[918 11]>
/app/chat-hook/hoon:<[915 3].[918 11]>
/app/chat-hook/hoon:<[914 3].[918 11]>
/app/chat-hook/hoon:<[107 25].[107 50]>
/app/chat-hook/hoon:<[107 9].[113 79]>
/app/chat-hook/hoon:<[106 9].[113 79]>
/app/chat-hook/hoon:<[105 9].[113 79]>
/app/chat-hook/hoon:<[103 9].[113 79]>
/app/chat-hook/hoon:<[102 9].[113 79]>
/app/chat-hook/hoon:<[101 9].[113 79]>
/app/chat-hook/hoon:<[100 7].[114 18]>
/app/chat-hook/hoon:<[99 7].[114 18]>
/app/chat-hook/hoon:<[91 7].[114 18]>
/app/chat-hook/hoon:<[90 5].[189 7]>
/app/chat-hook/hoon:<[88 5].[189 7]>
/app/chat-hook/hoon:<[87 5].[189 7]>
/app/chat-hook/hoon:<[86 5].[189 7]>
/app/chat-hook/hoon:<[85 5].[189 7]>
/app/chat-hook/hoon:<[84 5].[355 7]>
/app/chat-hook/hoon:<[83 5].[355 7]>
```

**Comment by matildepark** [comment]
Ooh, leftover case from a bad group-store migration perhaps.

—
~haddef-sigwen
https://urbit.org

**Comment by matildepark** [comment]
Haddef might have this as well, but it's harder to assess because of hosting. I will crack open Solaris after I force the merge, but I'm unsure OTA applied (or if I even have it yet).

**Comment by matildepark** [comment]
At this point — can I just have a one-liner to delete malformed chats without groups?

**Comment by Fang-** [comment]
So I poked around the state of these things a bunch. This was kinda painful, chat-hook has no scry endpoints.

In any case, interesting findings. Every locally hosted chat in chat-store has a matching group. For chat-hook however, these two do not have matching groups:

```hoon
{ /~palfun-foslup/dm--sipnes-hanhes
  /~palfun-foslup/dm--naltyc-wornes
}
```

But then, we look at what chats are in chat store, and what groups are in groups-store and we see these entries:

```hoon
{ /~palfun-foslup/dm--~sipnes-hanhes
  /~palfun-foslup/dm--~naltyc-wornes
}
```

Note the `~` in front of the names.

So there is (or at one point was) a bug where the `~` was included erroneously? Which is weird, you'd expect these paths to just get passed around as opaque values, instead of (re)constructed in multiple places. Not sure what's up with that, but userspace devs might be able to take a guess.

Kinda don't want to nuke these DMs, and we probably need to solve this for regular users anyway, so.. what's next?

**Comment by matildepark** [comment]
Even the initial new DM screen didn't include the sig (c162f0f) but maybe there was a specific period of time when this edge case was possible?

**Pull request merged**